modules/illumina.nf

```diff
@@ -171,9 +196,25 @@ process trimPrimerSequences {
     output:
     tuple val(sampleName), path("${sampleName}.mapped.bam"), path("${sampleName}.mapped.bam.bai"), emit: mapped
     tuple val(sampleName), path("${sampleName}.mapped.primertrimmed.sorted.bam"), path("${sampleName}.mapped.primertrimmed.sorted.bam.bai" ), emit: ptrim
+    tuple val(sampleName), path("${sampleName}_trimPrimerSequences_provenance.yml"), emit: provenance
 
     script:
     """
+    printf -- "- process_name: trimPrimerSequences\\n"                                             >> ${sampleName}_trimPrimerSequences_provenance.yml
+    printf -- "  tools:\\n"                                                                        >> ${sampleName}_trimPrimerSequences_provenance.yml
+    printf -- "    - tool_name: samtools\\n"                                                       >> ${sampleName}_trimPrimerSequences_provenance.yml
```

Comment:
I think we may want to list out each samtools subcommand as a separate tool. Something like:

```yml
tools:
  - tool_name: samtools
    subcommand: view
    parameters:
      ...
  - tool_name: samtools
    subcommand: index
    parameters:
      ...
  - tool_name: samtools
    subcommand: sort
    parameters:
      ...
```